src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -783,6 +781,36 @@ internal bool IsAutoProperty
                 return null;
             }
         }
+
+        private bool AllowInitializer
+        {
+            get
+            {
+                // PROTOTYPE(semi-auto-props): Fix implementation for semi auto properties.
+                return (_setMethod is null && _getMethod?.BodyShouldBeSynthesized == true) ||
+                    _setMethod?.BodyShouldBeSynthesized == true;
+            }
+        }
+
+        private bool AllowFieldAttributeTarget
+        {
+            get
+            {
+                // PROTOTYPE(semi-auto-props): Fix implementation for semi auto properties.
+                return (_setMethod is null && _getMethod?.BodyShouldBeSynthesized == true) ||
+                    _setMethod?.BodyShouldBeSynthesized == true;
```

Comment:
I am not sure if using the same implementation as for ```AllowInitializer``` is the right thing to do. Earlier I was suggesting the following: "Could this be as simple as checking if getter exists and is auto-implemented instead?" It feels like that logic will exactly match the name of the property that we are replacing, IsAutoPropertyWithGetAccessor. I could imagine even more relaxed implementation. Something like 
```
_getMethod?.BodyShouldBeSynthesized == true ||
                    _setMethod?.BodyShouldBeSynthesized == true
```

For an attribute, we just need to have a field. Requirements for an initializer are more strict. Especially taking [https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-01-12.md#conclusion](https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-01-12.md#conclusion) into consideration.

Reply:
> ```csharp
> _getMethod?.BodyShouldBeSynthesized == true ||
>                     _setMethod?.BodyShouldBeSynthesized == true
> ```

@AlekseyTs Both of the logics don't seem to take semi auto properties into account, unless I'm missing something. For example, `public string S { get => field; }` should be allowed to have a field attribute target. But I agree your logic is a bit more accurate.